### PR TITLE
UTY-1054: Resolve merge conflicts between master and feature branch

### PR DESCRIPTION
**Pull request by JonasImprobable**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Pull in and resolve all merge conflicts between master and feature/monobehaviour-support-shared. Merging this PR into feature/monobehaviour-support-shared allows us to merge feature/monobehaviour-support-shared into master without conflicts.
#### Tests
- Played around in playground
- Existing unit tests are passing
#### Documentation
How is this documented (for example: release note, upgrade guide, feature page, in-code documentation)?
#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: workers/unity/Packages/com.improbable.gdk.core/Logging/ILogDispatcher.cs [hunk 1]
```diff
@@ -1,10 +1,9 @@
-using System;
 using Improbable.Worker.Core;
 using UnityEngine;
 
 namespace Improbable.Gdk.Core
 {
-    public interface ILogDispatcher : IDisposable
```
Comment (MPeti01) [inline workers/unity/Packages/com.improbable.gdk.core/Logging/ILogDispatcher.cs]: This was added by Firtina for the cleanup on some MonoBehaviour tests to work, in this commit: https://github.com/spatialos/UnityGDK/commit/f9bdfdb85404fc0d1856b203dec72d7ea3e52198

So I think the merge should have the MonoBehaviour branch version being applied.

File: workers/unity/Assets/Generated/AssemblyInfo.cs.meta [hunk 2]
```diff
@@ -1,5 +1,5 @@
 fileFormatVersion: 2
-guid: a1081d047c7ae894f99b848b30a744a2
+guid: 2a2ee80ffd9d5af4e810a89cc9aa1fe4
```
[the remark this answers is not in the document]
Comment (JonasImprobable) [reply workers/unity/Assets/Generated/AssemblyInfo.cs.meta]: Yes, this is a new file introduced by the monobehaviour API feature. It allows the `Improbable.Gdk.Core.EditmodeTests` to access internal generated classes.

File: workers/unity/Packages/com.improbable.gdk.core/Logging/LoggingDispatcher.cs [hunk 3]
```diff
@@ -29,9 +29,5 @@ public void HandleLog(LogType type, LogEvent logEvent)
                     logEvent.Context);
             }
         }
-
-        public void Dispose()
```
Comment (MPeti01) [inline workers/unity/Packages/com.improbable.gdk.core/Logging/LoggingDispatcher.cs]: Same as above

File: workers/unity/Packages/com.improbable.gdk.core/Worker/Worker.cs [hunk 4]
```diff
@@ -74,7 +74,6 @@ public static void Disconnect(Worker worker)
 
         public void Dispose()
         {
-            LogDispatcher.Dispose();
```
Comment (MPeti01) [inline workers/unity/Packages/com.improbable.gdk.core/Worker/Worker.cs]: Same as above

File: workers/unity/Assets/Generated/AssemblyInfo.cs.meta (hunk 2 above)
[the remark this answers is not in the document]
Comment (jamiebrynes7) [reply workers/unity/Assets/Generated/AssemblyInfo.cs.meta]: Ah okay - that's fine.

We should look into building a test project for this sort of thing

File: workers/unity/Packages/com.improbable.gdk.core/Logging/ILogDispatcher.cs (hunk 1 above)
Comment (JonasImprobable) [reply workers/unity/Packages/com.improbable.gdk.core/Logging/ILogDispatcher.cs]: Interesting. THe tests are passing even without this change. Do you remember the context behind this change? Is this to ensure that Dispose() is called on ForwardingDispatchers?

Comment (JonasImprobable) [reply workers/unity/Packages/com.improbable.gdk.core/Logging/ILogDispatcher.cs]: Reverted changes. ILogDispatcher is an IDisposable again.

File: workers/unity/Packages/com.improbable.gdk.core/Logging/LoggingDispatcher.cs (hunk 3 above)
Comment (JonasImprobable) [reply workers/unity/Packages/com.improbable.gdk.core/Logging/LoggingDispatcher.cs]: Reverted.

File: workers/unity/Packages/com.improbable.gdk.core/Worker/Worker.cs (hunk 4 above)
Comment (JonasImprobable) [reply workers/unity/Packages/com.improbable.gdk.core/Worker/Worker.cs]: Reverted.

**Comment by JonasImprobable** [comment]
@jessicafalk adding you as a person from outside the monobehaviour team. Could you please sign of this change here: https://github.com/spatialos/UnityGDK/pull/245/commits/88e30302a43bbc34a678cf724f12eaaed36159f9 
It's not related to monoebahviours. It's a small change that we snuck in to fix tests. Just trying to minimise friction when merging everything back to master.

**Pull request merged**